entity-management-web/src/main/java/eu/europeana/entitymanagement/batch/processor/EntityUpdateProcessor.java

```diff
@@ -40,13 +42,28 @@ public EntityUpdateProcessor(EntityRecordService entityRecordService,
 
     @Override
     public EntityRecord process(@NonNull EntityRecord entityRecord) throws EuropeanaApiException {
+
+        EntityProxy europeanaProxy = entityRecord.getEuropeanaProxy();
+        EntityProxy externalProxy = entityRecord.getExternalProxy();
+        if (europeanaProxy == null || externalProxy == null) {
+            throw new EuropeanaApiException(String.format(
+                    "Unable to process entity record with id: %s. Europeana proxy or external proxy is not available in the record!",
+                    entityRecord.getEntityId()));
+        }
+
+        Entity europeanaEntity = europeanaProxy.getEntity();
+        Entity externalEntity = externalProxy.getEntity();
+
+        validateMinimalConstraints(externalEntity);
+        //data processing must be performed on a deep clone
+        Entity normalizedEntity = EntityObjectFactory.createNewEntity(externalEntity);
+        emEntityFieldCleaner.cleanAndNormalize(normalizedEntity);
+        Entity consolidatedEntity = entityRecordService.mergeEntities(europeanaEntity, externalEntity);
```

Comment:
Consolidated entity is supposed to contain normalised fields, so this line should be:

```
Entity consolidatedEntity = entityRecordService.mergeEntities(europeanaEntity, normalizedEntity);
```

Reply:
true, fixed